gnucash/gnome-utils/gnc-report-combo.c

```diff
@@ -0,0 +1,595 @@
+    }
+
+    if (!found)
+    {
+        GtkTreeIter iter;
+        QofBook *book = gnc_get_current_book ();
+        gchar *tmp;
+
+        if (priv->active_report_name)
+            tmp = g_strdup (priv->active_report_name);
+        else
+            tmp = g_strdup (_("Selected Report is Missing"));
+
+        gtk_list_store_prepend (GTK_LIST_STORE(model), &iter);
+        gtk_list_store_set (GTK_LIST_STORE(model), &iter,
+                            RC_NAME, tmp,
+                            RC_GUID, priv->active_report_guid,
+                            RC_MISSING, TRUE,
+                            -1);
+
+        g_free (tmp);
+        gtk_combo_box_set_active_iter (GTK_COMBO_BOX(priv->combo), &iter);
+    }
+    return found;
+}
+
+/********************************************************************
+ * update_invoice_list
+ *
+ * this procedure does the real work of displaying a sorted list of
+ * available invoice reports
+ ********************************************************************/
+static void
+update_report_list (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(priv->combo));
+
+    SCM template_menu_name = scm_c_eval_string ("gnc:report-template-menu-name/report-guid");
+    SCM get_rpt_guids = scm_c_eval_string (priv->rpt_guids);
+    SCM reportlist;
+    SCM rpt_guids;
+
+    gtk_tree_sortable_set_sort_column_id (GTK_TREE_SORTABLE(model),
+                                          RC_NAME, GTK_SORT_ASCENDING);
+
+    gtk_list_store_clear (GTK_LIST_STORE(model));
+
+    if (!scm_is_procedure (get_rpt_guids))
+        return;
+
+    reportlist = scm_call_0 (get_rpt_guids);
+    rpt_guids = reportlist;
+
+    if (scm_is_list (rpt_guids))
+    {
+        int i;
+        GtkTreeIter iter;
+
+        for (i = 0; !scm_is_null (rpt_guids); i++)
+        {
+            gchar *guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
+            gchar *name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name,
+                                                  SCM_CAR(rpt_guids), SCM_BOOL_F));
+
+            gtk_list_store_append (GTK_LIST_STORE(model), &iter);
+            gtk_list_store_set (GTK_LIST_STORE(model), &iter,
+                                RC_NAME, name,
+                                RC_GUID, guid_str,
+                                RC_MISSING, FALSE,
+                                -1);
+            g_free (name);
+            g_free (guid_str);
+
+            rpt_guids = SCM_CDR(rpt_guids);
+        }
+    }
+}
+
+static void
+update_warning_tooltip (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    gchar *tool_tip;
+
+    if (priv->active_report_name)
+        /* Translators: %s is the report name. */
+        tool_tip = g_strdup_printf (_("'%s' is missing"),
+                                      priv->active_report_name);
```

Comment:
"Report x is missing" may be misleading. From reading the gnc_report_combo_set_active function any report guid, name combination can be passed to this widget. Some of these combinations may be as you suggest be saved reports that no longer exist in the book, but it could just as well be a combination that refers to a report that's not a printable invoice report. So perhaps that's a useful second check to give a more precise warning message to the user.